libfive/include/libfive/oracle/oracle_clause.hpp

```diff
@@ -54,8 +54,20 @@ class OracleClause
     {
         return {};
     }
+
     unsigned rank() const;
 
+    virtual std::unique_ptr<OracleClause> clone() const = 0;
+
+    /*
+     * Returns the absolute value of the oracle.
+     * This can used to improve performance on certain oracles.
+     */
+    virtual libfive::Tree abs() const {
+      Tree tree(clone());
```

Comment:
I think this can be this can be 
```c++
virtual libfive::Tree abs(const libfive::Tree& self) const {
      return abs(self);
}
```

Reply:
This seems a bit hacky since it’s kind of a static method but it’s not so we can override it and it has this additional argument that should not actually be needed :/ I don’t have a strong opinion on this though, so if you think we should go with this I can change it :)

Reply:
If you are ok with this then lets go with this. One other way (which I kind of like the best). Is where you just create another interface instead of adding to OracleClause. call it something like.  

```c++
class CustomAbsOracle {
  virtual libfive::Tree abs() const = 0;
}

```
and then just have the bvhOracleClause implement this. You also dont have to modify libfive with this interface

Reply:
yes I like this approach, lets go with this :)